indico/modules/events/management/controllers/cloning.py

```diff
@@ -101,21 +104,34 @@ class PatternCloneCalculator(CloneCalculator):
     def _calculate(self, form):
         args = {'dtstart': self._naivify(form.start_dt.data)}
         args.update(self._calc_stop_criteria(form))
-        return self._tzify(rrule.rrule(rrule.MONTHLY,
-                                       interval=form.num_months.data,
-                                       byweekday=form.week_day.week_day_data,
-                                       bysetpos=form.week_day.day_number_data,
-                                       **args))
+        dates = self._tzify(rrule.rrule(rrule.MONTHLY,
+                                        interval=form.num_months.data,
+                                        byweekday=form.week_day.week_day_data,
+                                        bysetpos=form.week_day.day_number_data,
+                                        **args))
+        return dates, False
 
 
 class IntervalCloneCalculator(CloneCalculator):
     form_class = CloneRepeatIntervalForm
 
     def _calculate(self, form):
-        args = {'dtstart': self._naivify(form.start_dt.data)}
-        args.update(self._calc_stop_criteria(form))
+        last_day_of_month = False
         freq, interval = relativedelta_to_rrule_interval(form.recurrence.data)
-        return self._tzify(rrule.rrule(freq, interval=interval, **args))
+        # check if last day of month
+        dtstart = self._naivify(form.start_dt.data)
+        next_day = dtstart + timedelta(days=1)
+        if freq == rrule.MONTHLY and next_day.day == 1:
+            args = {'dtstart': next_day}
+            args.update(self._calc_stop_criteria(form))
+            dates = rrule.rrule(freq, interval=interval, **args)
+            last_day_of_month = True
+            dates = self._tzify([date - timedelta(days=1) for date in dates])
+        else:
+            args = {'dtstart': dtstart}
+            args.update(self._calc_stop_criteria(form))
+            dates = self._tzify(rrule.rrule(freq, interval=interval, **args))
+        return dates, last_day_of_month
```

Comment:
i'd still get rid of the flag and have to return statements (`return dates, True`  and `return dates, False`).

in any case, if you prefer to keep it like this i would move the `last_day_of_month = False` into the `else` block